clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -369,6 +369,71 @@ private static DefaultRecord readFrom(ByteBuffer buffer,
         }
     }
 
+    /**
+     * Reading data from DataInput whithout create a new ByteBuffer.
+     *
+     * @param input
+     * @param baseOffset
+     * @param baseTimestamp
+     * @param baseSequence
+     * @param logAppendTime
+     * @return
+     * @throws IOException
+     */
+    public static SimplifiedDefaultRecord simplifiedreadFrom(DataInput input,
```

Comment:
How about rename to `readFromSkipKV` to be more specific for other readers? See my other comment below.

Ditto below.